src/patient-registration/patient-registration.resource.tsx

```diff
@@ -22,5 +23,56 @@ export function getUniquePatientIdentifier(abortController: AbortController) {
   return openmrsFetch('/module/idgen/generateIdentifier.form?source=1');
 }
 
+export function getPrimaryIdentifierType(abortController: AbortController) {
+  return openmrsFetch('/ws/rest/v1/metadatamapping/termmapping?v=full&code=emr.primaryIdentifierType', {
+    signal: abortController.signal,
+  }).then(response => {
+    return openmrsFetch('/ws/rest/v1/patientidentifiertype/' + response.data.results[0].metadataUuid, {
+      signal: abortController.signal,
+    }).then(response => ({
+      name: response.data.name,
+      fieldName: camelCase(response.data.name),
+      required: response.data.required,
+      uuid: response.data.uuid,
+      format: response.data.format,
```

Comment:
This is an interesting choice—why not just 
```
    }).then(response => response.data)
```

?

Reply:
I intended to return an instance of a [PatientIdentifierType](https://github.com/samuelmale/openmrs-esm-patient-registration/blob/MF-217/src/patient-registration/patient-registration-helper.tsx#L34-L40). Seems I'm doing something superfluous due to my limited JavaScript knowledge?

Reply:
Oh sure, yeah. But the thing with the type-checking is that extra properties are allowed. You can treat `response.data` as a PatientIdentifierType object even if it has extra stuff on it. Does that make sense?

Reply:
Yes it does, this means the JS engine will do the type casting for us.

Reply:
It would work but JS's smart Type casting may not infer [this property](https://github.com/samuelmale/openmrs-esm-patient-registration/blob/MF-217/src/patient-registration/patient-registration.resource.tsx#L34) for us automatically: 

```javascript

then(response => ({
      .......
      fieldName: camelCase(response.data.name),
      .......
    }));

```

Reply:
Yeah, sort of—JS doesn't really do casting between objects at all. The object just gets passed through, and the receiving code tries to use it.

So yeah, you're absolutely right that the resulting object would not magically camelCase the `fieldName` property. You could camelCase at the time of usage, which might make the code simpler, and also make clearer _why_ the field is being camelCased in the first place. Or you could do

```javascript
    }).then(response => ({
      fieldName: camelCase(response.data.name),
      ...response.data
```

The only reason to write the code the way you've written it would be if the `response.data` object is many times larger than the data you intend to use. Which may be valid, but probably isn't worth worrying about right now.